zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Note.java

```diff
@@ -55,9 +59,23 @@
 public class Note implements Serializable, ParagraphJobListener {
   private static final Logger logger = LoggerFactory.getLogger(Note.class);
   private static final long serialVersionUID = 7920699076577612429L;
-  private static final Gson gson = new GsonBuilder()
-      .registerTypeAdapterFactory(Input.TypeAdapterFactory)
-      .create();
+  private static Gson gson = new GsonBuilder()
+      .setPrettyPrinting()
+      .registerTypeAdapterFactory(new NotebookTypeAdapterFactory<Paragraph>(Paragraph.class) {
+        @Override
+        protected void beforeWrite(Paragraph source, JsonElement toSerialize) {
+          Map<String, ParagraphRuntimeInfo> runtimeInfos = source.getRuntimeInfos();
+          if (runtimeInfos != null) {
+            JsonElement jsonTree = gson.toJsonTree(runtimeInfos);
+            if (toSerialize instanceof JsonObject) {
+              JsonObject jsonObj = (JsonObject) toSerialize;
+              jsonObj.add("runtimeInfos", jsonTree);
+            }
+          }
+        }
+      }).setDateFormat("yyyy-MM-dd HH:mm:ss.SSS")
```

Comment:
I only see it is used in Note before, but I believe it is OK for gson in other classes use the same date format and register the same type adapter. But setPrettyPrinting is not true for all the classes.